tt_eager/tt_dnn/op_library/sharded/multi_core/sharded_op_multi_core.cpp

```diff
@@ -467,15 +490,22 @@ operation::ProgramWithCallbacks sharded_to_interleaved_multi_core(const Tensor&
             }
         }
     }
-    auto override_runtime_arguments_callback = [unary_reader_kernel_id, unary_writer_kernel_id, cb_src0, cores](
+    auto override_runtime_arguments_callback = [unary_reader_kernel_id, unary_writer_kernel_id, cb_src0, cores, num_slices](
                                                    const void* operation,
                                                    Program& program,
                                                    const std::vector<Tensor>& input_tensors,
                                                    const std::vector<std::optional<const Tensor>>&,
                                                    const std::vector<Tensor>& output_tensors) {
         auto src_buffer = input_tensors.at(0).buffer();
 
-        auto dst_buffer = output_tensors.at(0).buffer();
+        Buffer* dst_buffer = nullptr;
+        if (num_slices > 1) {
+            // If we have num_slices > 1, it means that our op is S->I partial.
+            // And currently we store output tensors there as input[1]
```

Comment:
Why we are storing tensors in inputs_tensors.at(1)?

Reply:
That's how it's done in a similar op when you need to take in optional output tensor